conrod_core/src/widget/primitive/shape/polygon.rs

```diff
@@ -271,18 +279,10 @@ impl<I> Colorable for Polygon<I> {
 pub fn triangles<I>(points: I) -> Option<Triangles<I::IntoIter>>
     where I: IntoIterator<Item=Point>,
 {
-    let mut points = points.into_iter();
-    let first = match points.next() {
-        Some(p) => p,
-        None => return None,
-    };
-    let prev = match points.next() {
-        Some(p) => p,
-        None => return None,
-    };
+    let points = points.into_iter();
     Some(Triangles {
-        first: first,
-        prev: prev,
+        first: [0.0,0.0],
+        prev: [0.0,0.0],
```

Comment:
Are these `first` and `prev` fields still used? Could they possibly be removed now that we retrieve three new points each time we yield a triangle? Or are they still used elsewhere?